### PR TITLE
adding download api to launch client, updating example

### PR DESCRIPTION
**Pull request by ian-scale**

### CODE REVIEW

**Comment by yunfeng-scale** [comment]
can you share screenshot of newly added page?

File: clients/python/llmengine/model.py [hunk 1]
```diff
@@ -366,3 +368,50 @@ def delete(cls, model: str) -> DeleteLLMEndpointResponse:
...
+        The user can then download these urls to obtain the fine-tuned model. If called 
+        on a nonexistent model, an error will be thrown.
+
+        Args:
+            model_name (`str`):
+                name of the fine-tuned model 
+            download_format (`str`): 
+                download format requested (default=huggingface) 
+        Returns:
+            DownloadModelResponse: an object that contains a dictionary of filenames, urls from which to download the model weights
```
Comment (yunfeng-scale) [inline clients/python/llmengine/model.py:393]: can you mention those urls are presigned s3 urls and what's the expiration?

Comment (ian-scale) [reply clients/python/llmengine/model.py:393]: yes I'll make that explicit!

**Comment by ian-scale** [comment]
> can you share screenshot of newly added page?

i.e. the .ipynb example?

File: clients/python/llmengine/model.py [hunk 2]
```diff
@@ -366,3 +368,51 @@ def delete(cls, model: str) -> DeleteLLMEndpointResponse:
         """
         response = cls._delete(f"v1/llm/model-endpoints/{model}", timeout=DEFAULT_TIMEOUT)
         return DeleteLLMEndpointResponse.parse_obj(response)
+
+    @classmethod
+    def download(
+        cls,
+        model_name: str,
+        download_format: str = "huggingface",
```
Comment (yixu34) [inline clients/python/llmengine/model.py:376]: It seems like we've used `hugging_face` elsewhere in LLM Engine. I think the motivation for this was due to Hugging Face being two words?
image

File: clients/python/llmengine/model.py [hunk 3]
```diff
@@ -366,3 +368,51 @@ def delete(cls, model: str) -> DeleteLLMEndpointResponse:
...
+        on a nonexistent model, an error will be thrown.
+
+        Args:
+            model_name (`str`):
+                name of the fine-tuned model
+            download_format (`str`):
+                download format requested (default=huggingface)
+        Returns:
+            DownloadModelResponse: an object that contains a dictionary of filenames, urls from which to download the model weights.
+            The urls are presigned urls that grant temporary access and expire after an hour.
```
Comment (yixu34) [inline clients/python/llmengine/model.py:394]: Think this is OK for now, but not sure if this is technically true in all contexts, e.g. self-hosting.

**Review by yixu34 — COMMENTED** [review]
Normally I'd suggest adding a helper function like `Model.download` instead of relying on these code snippets, but given that we don't actually want to make this _too_ easy, I think that's fine.

File: clients/python/llmengine/model.py (hunk 3 above)
Comment (ian-scale) [reply clients/python/llmengine/model.py:394]: This make sense, I'm actually not exactly sure what the behavior would be in a self-hosted context. We can think about this more as it becomes relevant in the future?

File: clients/python/llmengine/model.py (hunk 2 above)
Comment (ian-scale) [reply clients/python/llmengine/model.py:376]: oh okay, that's a relatively easy fix (currently we're not actually doing anything with the format token anyway)

**Pull request merged**